common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java

```diff
@@ -143,15 +143,25 @@ public static synchronized PooledByteBufAllocator getSharedPooledByteBufAllocato
 
   public static PooledByteBufAllocator getPooledByteBufAllocator(
       TransportConf conf, AbstractSource source, boolean allowCache) {
+    return getPooledByteBufAllocator(conf, source, allowCache, 0);
+  }
+
+  public static PooledByteBufAllocator getPooledByteBufAllocator(
+      TransportConf conf, AbstractSource source, boolean allowCache, int coreNum) {
     if (conf.getCelebornConf().networkShareMemoryAllocator()) {
       return getSharedPooledByteBufAllocator(
           conf.getCelebornConf(),
           source,
           allowCache && conf.getCelebornConf().networkMemoryAllocatorAllowCache());
     }
+    int arenas;
+    if (coreNum != 0) {
+      arenas = coreNum;
+    } else {
+      arenas = conf.getCelebornConf().networkAllocatorArenas();
```

Comment:
Could this limit the arena with netty max thread num?

Reply:
Should we limit this max value in `conf.getCelebornConf().networkAllocatorArenas()` or just in this un-shared pool memory allocator case?

Reply:
User environments are various, there won't be a proper limit for all environments. I think this should have no limit.